pkg/mobile/mobileClient.go

```diff
@@ -66,3 +76,73 @@ func (r *MobileClientRepoImpl) DeleteByName(name string) error {
 func (r *MobileClientRepoImpl) Create(app *v1alpha1.MobileClient) error {
 	return sdk.Create(app)
 }
+
+type MobileWatcher struct {
+	events chan watch.Event
+}
+
+func (w MobileWatcher) Stop() {
+	done := make(chan struct{})
+	go func() {
+		drain := true
+		for drain {
+			select {
+			case <-done:
+				drain = false
+			case <-w.events:
+			case <-time.After(1 * time.Second):
+			}
+		}
+	}()
+	m.Lock()
+	handler.RemoveWatcher(w)
+	m.Unlock()
```

Comment:
is this because a race condition will happen when the "Stop" method is called on multiple watchers? if so would a map a better option to store the watchers? This way it will be more efficient as you don't need to loop through the array to find the watcher and build a new array.

Reply:
The race condition will probably happen also for multiple `Stop`s. I saw the issue when stopping one watcher and sending notifications in same time. Basically any read/write operations happening simultaneously will cause it. So I think that map would not help here.

Reply:
A normal golang map won't help, but there is a special type of [Map in the sync package](https://golang.org/pkg/sync/#Map) which will help. Probably worth taking a look.

Reply:
thanks, will take a look